cvmfs/sync_mediator.h

```diff
@@ -50,6 +51,49 @@ struct Counters;
 
 namespace publish {
 
+class SyncDiffReporter : public DiffListener {
+ public:
+  explicit SyncDiffReporter(bool print_dots,
+                            unsigned int processing_dot_interval = 100)
+      : print_dots_(print_dots),
+        processing_dot_interval_(processing_dot_interval),
+        changed_items_(0) {}
+
+  void OnInit(const history::History::Tag &from_tag,
```

Comment:
Since the methods overwrite, we should mark them as virtual, e.g. `virtual void OnInit(...`